pkg/kubelet/kubelet.go

```diff
@@ -1827,7 +1833,14 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, podStatus *kubecont
 	}
 
 	// Generate final API pod status with pod and status manager status
-	apiPodStatus := kl.generateAPIPodStatus(pod, podStatus)
+	// by default, we use the internal status generation func, but we defer to
+	// the override function if specified.  this is used for eviction scenarios
+	// that interacted directly with pod workers to update pod status.
+	generateAPIPodStatus := kl.generateAPIPodStatus
+	if o.podStatusFunc != nil {
+		generateAPIPodStatus = o.podStatusFunc
```

Comment:
I still need a mechanism to denote this is where the status should be taken from.

If I add a sync type "kill", and a "killStatusFunc' maybe that will be clearer.